Dockerfile

```diff
@@ -1,32 +1,64 @@
-ARG EXPERIMENTS_VERSION
+FROM faasm/cpp-sysroot:0.0.26
 
-# We pin to a specific sysroot release
-FROM faasm/cpp-sysroot:0.0.22 as build-step
+# Download and install OpenMPI
+WORKDIR /tmp
+RUN wget https://download.open-mpi.org/release/open-mpi/v4.1/openmpi-4.1.0.tar.bz2
+RUN tar xf openmpi-4.1.0.tar.bz2
+WORKDIR /tmp/openmpi-4.1.0
+RUN ./configure --prefix=/usr/local
+RUN make -j `nproc`
+RUN make install
+# The previous steps take a lot of time, so don't move these lines and beneffit
+# from Docker's incremental build
+
+# Add an mpirun user
+ENV USER mpirun
+RUN adduser --disabled-password --gecos "" ${USER} && \
+    echo "${USER} ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers
+RUN apt update
+# Dev tools delete eventually
+RUN apt install -y gdb vim
+
+# Set up SSH (for native MPI)
+RUN apt update && apt upgrade -y
+RUN apt install -y openssh-server
+RUN mkdir /var/run/sshd
+RUN echo 'root:${USER}' | chpasswd
+RUN sed -i 's/PermitRootLogin without-password/PermitRootLogin yes/' /etc/ssh/sshd_config
+RUN sed 's@session\s*required\s*pam_loginuid.so@session optional pam_loginuid.so@g' -i /etc/pam.d/sshd
+
+# User SSH config
+ENV HOME /home/${USER}
+WORKDIR ${HOME}/.ssh
+COPY ./ssh/config config
+COPY ./ssh/id_rsa.mpi id_rsa
+COPY ./ssh/id_rsa.mpi.pub id_rsa.pub
+COPY ./ssh/id_rsa.mpi.pub authorized_keys
+RUN ssh-keygen -A
+RUN chmod -R 600 ${HOME}/.ssh*
+RUN chmod 700 ${HOME}/.ssh
+RUN chmod 644 ${HOME}/.ssh/id_rsa.pub
+RUN chmod 664 ${HOME}/.ssh/config
+RUN chown -R ${USER}:${USER} ${HOME}/.ssh
```

Comment:
All this SSH/ MPI stuff is from the `experiments-base` image, but is only used in this project, so it makes sense to inline it for now so we don't have to build two images for this experiment.